extensions/wikia/Discussions/maintenance/SiteStatus.class.php

```diff
@@ -0,0 +1,411 @@
+			return '';
+		}
+		return $this->firstPostFound->format('Y-m-d H:i:s');
+	}
+
+	private function probeSiteAvailable() {
+		$this->debug( "Probing wiki site" );
+
+		$result = ( new \WikiaSQL() )
+			->SELECT( "city_flags" )
+			->FROM( self::TABLE_CITY_LIST )
+			->WHERE( 'city_id' )->EQUAL_TO( $this->siteId )
+				->AND_( 'city_public' )->EQUAL_TO( 1 )
+			->run(
+				$this->getCentralDbr(),
+				function(  $result ) {
+					/** @var \ResultWrapper|bool $result */
+					if ( !is_object( $result ) ) {
+						return false;
+					}
+
+					$row = $result->fetchObject();
+					if ( empty( $row ) ) {
+						return false;
+					}
+
+					// Flag zero means no problem
+					if ( $row->city_flags == 0 ) {
+						return true;
+					}
+
+					// If there are flags, make sure this wiki isn't marked as free
+					return !( $row->city_flags & \WikiFactory::FLAG_FREE_WIKI_URL );
+				},
+				false
+			);
+
+		// Map this to an int
+		$this->siteAvailable = $result ? 1 : 0;
+	}
+
+	private function updateWikiVariables() {
+		$this->debug( "Updating wiki variables." );
+
+		$this->forumsEnabled = $this->getVariableValue( 'wgEnableForumExt' );
+		$this->discussionsEnabled = $this->getVariableValue( 'wgEnableDiscussions' );
+		$this->navigationEnabled = $this->getVariableValue( 'wgEnableDiscussionsNavigation' );
+	}
+
+	private function findExistingPosts() {
+		$this->debug( "Finding existing posts" );
+
+		$this->findExistingThreadedForumPosts();
+		$this->findExistingWikiForumPosts();
+	}
+
+	private function findExistingThreadedForumPosts() {
+		$num = ( new \WikiaSQL() )
+			->SELECT( "count(*) " )->AS_( "num_posts" )
+			->FROM( self::TABLE_COMMENTS )
+				->LEFT_JOIN( self::TABLE_PAGE )
+				->ON( 'page_id', 'comment_id' )
+			->WHERE( 'parent_comment_id' )->EQUAL_TO( 0 )
+				->AND_( 'archived' )->EQUAL_TO( 0 )
+				->AND_( 'deleted' )->EQUAL_TO( 0 )
+				->AND_( 'removed' )->EQUAL_TO( 0 )
+				->AND_( 'page_namespace' )->EQUAL_TO( NS_WIKIA_FORUM_BOARD_THREAD )
+			->run(
+				$this->localDbh,
+				function ( $result ) {
+					/** @var \ResultWrapper|bool $result */
+					if ( !is_object( $result ) ) {
+						return 0;
+					}
+
+					$row = $result->fetchObject();
+					return $row ? $row->num_posts : 0;
+				},
+				0
+			);
+
+		$this->debug("\tfound $num threaded forum posts" );
+		$this->numThreadedForumPosts = $num;
+	}
+
+	private function findExistingWikiForumPosts() {
+		$num = ( new \WikiaSQL() )
+			->SELECT( "count(*) " )->AS_( "num_posts" )
+			->FROM( self::TABLE_PAGE )
+			->WHERE( 'page_namespace' )->EQUAL_TO( NS_FORUM )
+			->run(
+				$this->localDbh,
+				function ( $result ) {
+					/** @var \ResultWrapper|bool $result */
+					if ( !is_object( $result ) ) {
+						return 0;
+					}
+
+					$row = $result->fetchObject();
+
+					return $row ? $row->num_posts : 0;
+				},
+				0
+			);
+
+		$this->debug("\tfound $num wiki forum posts" );
+		$this->numWikiForumPosts = $num;
+	}
+
+	private function probeDiscussions() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)